### PR TITLE
Logging: Drop Settings from security logger get calls

### PR DESCRIPTION
**Pull request by nik9000**
`Settings` is no longer required to get a `Logger` and we went to quite
a bit of effort to pass it to the `Logger` getters. This removes the
`Settings` logger lookups from the security and x-pack:core.

### CODE REVIEW

**Comment by cbuescher** [comment]
This must have fallen into some CI downtime, so no jenkins run was triggered yet, so: 
@elasticmachine test this please

**Review by cbuescher — APPROVED** [review]
@nik9000 I left a tiny question but in general LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/transform/chain/ChainTransformFactory.java [hunk 1]
```diff
@@ -20,8 +19,8 @@
 
     private final TransformRegistry registry;
 
-    public ChainTransformFactory(Settings settings, TransformRegistry registry) {
-        super(Loggers.getLogger(ExecutableChainTransform.class, settings));
+    public ChainTransformFactory(TransformRegistry registry) {
+        super(Loggers.getLogger(ExecutableChainTransform.class));
```
[the remark this answers is not in the document]
Comment (nik9000) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/transform/chain/ChainTransformFactory.java]: I should have done it here, yeah. The advantage of the latter is that the former is deprecated. It mostly just delegates to the latter.

**Comment by nik9000** [comment]
@elasticmachine retest this please

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**